src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -318,7 +318,7 @@ public DbConnection Close()
 		/// </summary>
 		public override void AfterTransactionCompletion(bool success, ITransaction tx)
 		{
-			using (new SessionIdLoggingContext(SessionId))
+			using (BeginContext())
```

Comment:
As already written, transaction completion may occur on closed session, so just setting the session id here.